core/includes/form.inc

```diff
@@ -4937,14 +4945,17 @@ function theme_form_element_label($variables) {
 /**
  * Sets a form element's class attribute.
  *
- * Adds 'required' and 'error' classes as needed.
+ * Also adds 'required' and 'error' classes/attributes as needed.
  *
  * @param $element
  *   The form element.
  * @param $name
  *   Array of new class names to be added.
+ *
+ * @since 1.??.? function renamed from _form_set_class() to
+ *   _form_set_attributes() since we're setting non-class attributes too.
```

Comment:
This was done in Drupal in a follow-up: https://www.drupal.org/project/drupal/issues/1503982 ([commit](https://git.drupalcode.org/project/drupal/-/commit/d3728cbf1bb7149aee983ede48f6807b165420cc))

Reply:
I'd love to see it marked as deprecated and a wrapper function added, even though this should be safe to change. It will make lives easier for anyone using this currently.

Reply:
I don't think we should create a wrapper and mark it as deprecated. It's not an API function so nothing outside of this file should be using it. We've got to draw a line somewhere.